### ISSUE TITLE
Refactor /osm as a separate lib

### ISSUE THREAD

**Issue by nyurik** [openstreetmap/iD#4409]
Would it make sense to refactor the OSM API XML parsing & constructing code in the `osm/` dir into a separate library?  It seems it would be highly useful to anyone who may want to implement a custom OSM editor for a specific task.  Thanks!

**Comment by bhousel** [comment]
I agree, I'd like to break iD into smaller library-like pieces and cut down on the interdependencies so that people can reuse the code for other things.

For now, you probably could pull in the entire iD project as a `devDependency` in another project and just import the modules that the custom app needs.  This is not straightforward to do, but should work.  If you bundle it with rollup, you might even be able to tree-shake out the parts of iD that aren't really used for anything (but I'm betting that the "cycles" in the iD code make this impossible).

**Comment by nyurik** [comment]
@bhousel I just found [angular-osm project](https://github.com/toutpt/angular-osm/tree/master/src/api). The lib is a small and simple OSM API module.  I am thinking of factoring it out into a separate project, and we could use it for various projects, possibly even migrating iD to use it?  The library uses [x2js](https://github.com/abdmob/x2js) to convert JSON to XML and back.

**Comment by bhousel** [comment]
> I just found angular-osm project. The lib has a small and simple OSM API module. I am thinking of factoring it out into a separate project, and we could use it for various projects, possibly even migrating iD to use it? The library uses x2js to convert JSON to XML and back.

That's good, it might work for what you are building 👍

**Comment by nyurik** [comment]
Do you think it makes sense to use it for iD?  Might not be - iD's API-related code is probably fairly stable by this point.

**Comment by bhousel** [comment]
I don’t think it makes sense to replace what we have now.

**Comment by zlant** [comment]
I dreamed of such a library for working with osm api and data modification functions (for example, cutting lines) when the [parking-lanes editor](https://github.com/zlant/parking-lanes) wrote.
